softlayer/data_source_softlayer_vlan.go

```diff
@@ -0,0 +1,74 @@
+package softlayer
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/softlayer/softlayer-go/filter"
+	"github.com/softlayer/softlayer-go/services"
+	"github.com/softlayer/softlayer-go/session"
+)
+
+func dataSourceSoftLayerVlan() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceSoftLayerVlanRead,
+
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+
+			"name": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				ConflictsWith: []string{"number", "router_hostname"},
+			},
+
+			"number": {
+				Type:          schema.TypeInt,
+				Optional:      true,
+				ConflictsWith: []string{"name"},
+			},
+
+			"router_hostname": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				ConflictsWith: []string{"name"},
+			},
+		},
+	}
+}
+
+func dataSourceSoftLayerVlanRead(d *schema.ResourceData, meta interface{}) error {
+	sess := meta.(*session.Session)
+	service := services.GetAccountService(sess)
+
+	name := d.Get("name").(string)
+	number := d.Get("number").(int)
+	routerHostname := d.Get("router_hostname").(string)
+
+	if name != "" {
+		networkVlans, err := service.
+			Mask("id").
+			Filter(filter.Path("networkVlans.name").Eq(name).Build()).
+			GetNetworkVlans()
+		if err != nil || len(networkVlans) == 0 {
+			return fmt.Errorf("Error obtaining VLAN id: %s", err)
+		}
+
+		d.SetId(fmt.Sprintf("%d", *networkVlans[0].Id))
```

Comment:
Why? these are not computed fields. The idea is to get the VLAN id which will be used in the other resoruces (lb_vpx, virtual_guest, bare_metal, scale_group). But what resource will be in need of number/router_hostname from providing the name in the data source?